psa-crypto-sys/src/utils.rs

```diff
@@ -0,0 +1,15 @@
+// Copyright 2020 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+#![allow(non_snake_case)]
+/// Additional functionality required that Mbed Crypto does not provide
+use crate::types::psa_algorithm_t;
+/// Retrieves the tag length from an aead_alg.
+/// Note: `aead_alg` is an AEAD algorithm, such that `PSA_ALG_IS_AEAD(aead_alg)` is `true`.
+pub fn PSA_ALG_AEAD_TAG_TRUNCATED_LENGTH(aead_alg: psa_algorithm_t) -> usize {
+    const TAG_LENGTH_MASK: u8 = 0b00111111; // tag lengths are 6 bits in length
+    const PSA_V1_0_0_TAG_LENGTH_START_BIT: u32 = 16; // tag length at bit position [21:16]
+
+    let pre_mask_tag_length = aead_alg >> PSA_V1_0_0_TAG_LENGTH_START_BIT;
+
+    (pre_mask_tag_length as u8 & TAG_LENGTH_MASK) as usize
```

Comment:
Why is there a need for converting to `u8` since you convert to `usize` immediately after? I think just `& TAG_LENGTH_MASK` is enough to keep only the lowest 6 bits set.

Reply:
Originally had that but compiler said no - 
```
14 |     (pre_mask_tag_length & TAG_LENGTH_MASK) as usize
   |                            ^^^^^^^^^^^^^^^ expected `u32`, found `u8`

```

Reply:
And if you make `const TAG_LENGTH_MASK: u32 = 0b111111` ?

Reply:
That works